controllers/partition_controller.go

```diff
@@ -0,0 +1,313 @@
+	}
+
+	if replicas != int32(len(podList.Items)) {
+		log.Info("replicas is different from the expected number of Pods", "expected", replicas, "actual", len(podList.Items))
+		return false, nil
+	}
+
+	nextRolloutTarget := r.nextRolloutTargetIndex(sts)
+	if nextRolloutTarget < 0 {
+		return false, nil
+	}
+
+	if podList.Items[nextRolloutTarget].Labels[appsv1.ControllerRevisionHashLabelKey] == sts.Status.UpdateRevision {
+		return true, nil
+	}
+
+	// If not all Pods are ready, the MySQLCluster becomes Unhealthy.
+	// Even if the MySQLCluster is not healthy, the rollout continues if the rollout target Pod is not ready.
+	// This is because there is an expectation that restarting the Not Ready Pod might improve its state.
+	if podutils.IsPodReady(&podList.Items[nextRolloutTarget]) && !r.isMySQLClusterHealthy(cluster) {
+		log.Info("MySQLCluster is not healthy", "name", cluster.Name, "namespace", cluster.Namespace)
+		return false, nil
+	}
+
+	ready := r.areAllChildPodsRolloutReady(ctx, sts, podList)
+
+	return ready, nil
+}
+
+// getSortedPodList returns a sorted child pod list.
+// The list is sorted by pod name with ascending order.
+func (r *StatefulSetPartitionReconciler) getSortedPodList(ctx context.Context, sts *appsv1.StatefulSet) (*corev1.PodList, error) {
+	podList := &corev1.PodList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(sts.Namespace),
+		client.MatchingLabels(sts.Spec.Selector.MatchLabels),
+	}
+
+	err := r.Client.List(ctx, podList, listOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	sort.Slice(podList.Items, func(i, j int) bool {
+		return podList.Items[i].Name < podList.Items[j].Name
+	})
+
+	return podList, nil
+}
+
+func (r *StatefulSetPartitionReconciler) areAllChildPodsRolloutReady(ctx context.Context, sts *appsv1.StatefulSet, sortedPodList *corev1.PodList) bool {
+	log := crlog.FromContext(ctx)
+
+	nextRolloutTarget := r.nextRolloutTargetIndex(sts)
+
+	// Proceed with the rollout for the next Pod to be rolled out, even if it is not Ready.
+	// We expect that the Pod'excludeNextRolloutTagetPodList state will improve by being updated through the rollout.
+	// All other Pods must be Ready.
+	for i, pod := range sortedPodList.Items {
+		if i == nextRolloutTarget {
+			continue
+		}
+		if pod.DeletionTimestamp != nil {
+			log.Info("Pod is in the process of being terminated", "name", pod.Name, "namespace", pod.Namespace)
+			return false
+		}
+		if !podutils.IsPodAvailable(&pod, 5, metav1.Now()) {
+			log.Info("Pod is not ready", "name", pod.Name, "namespace", pod.Namespace)
+			return false
+		}
+		for _, c := range pod.Status.InitContainerStatuses {
+			log.Info("Container is not ready", "pod", pod.Name, "namespace", pod.Namespace, "container", c.Name)
+			if !c.Ready {
+				return false
+			}
+		}
+		for _, c := range pod.Status.ContainerStatuses {
+			log.Info("Container is not ready", "pod", pod.Name, "namespace", pod.Namespace, "container", c.Name)
+			if !c.Ready {
+				return false
+			}
```

Comment:
Please change the field names in the log.
The `name` and `namespace` fields are duplicateed.

```
{"level":"info","ts":"2024-08-15T08:59:15Z","msg":"Pod is not ready","controller":"statefulset","controllerGroup":"apps","controllerKind":"StatefulSet","StatefulSet":{"name":"moco-test","namespace":"test"},"namespace":"test","name":"moco-test","reconcileID":"26c9b749-9edf-4904-9bf0-0cfb38156b4c","name":"moco-test-1","namespace":"test"}
{"level":"info","ts":"2024-08-15T08:59:25Z","msg":"Container is not ready","controller":"statefulset","controllerGroup":"apps","controllerKind":"StatefulSet","StatefulSet":{"name":"moco-test","namespace":"test"},"namespace":"test","name":"moco-test","reconcileID":"b8c0118e-83dd-488d-9569-d96724b418db","pod":"moco-test-1","namespace":"test","container":"copy-moco-init"}
```

Please also ckeck the other logs.